cli/src/claim_invitation.rs

```diff
@@ -0,0 +1,219 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use clap::Args;
+use std::{path::PathBuf, sync::Arc};
+
+use libparsec::{
+    get_default_config_dir,
+    internal::{
+        claimer_retrieve_info, DeviceClaimFinalizeCtx, DeviceClaimInProgress1Ctx,
+        DeviceClaimInProgress2Ctx, DeviceClaimInProgress3Ctx, DeviceClaimInitialCtx,
+        UserClaimFinalizeCtx, UserClaimInProgress1Ctx, UserClaimInProgress2Ctx,
+        UserClaimInProgress3Ctx, UserClaimInitialCtx, UserOrDeviceClaimInitialCtx,
+    },
+    BackendInvitationAddr, ClientConfig, DeviceAccessStrategy,
+};
+
+use crate::utils::*;
+
+#[derive(Args)]
+pub struct ClaimInvitation {
+    /// Parsec config directory
+    #[arg(short, long, default_value_os_t = get_default_config_dir())]
+    config_dir: PathBuf,
+    /// Device slughash
+    #[arg(short, long)]
+    device: Option<String>,
+    /// Server invitation address (e.g.: parsec://127.0.0.1:41905/Org?no_ssl=true&action=claim_user&token=4e45cc21e7604af196173ff6c9184a1f)
+    #[arg(short, long)]
+    addr: BackendInvitationAddr,
+}
+
+/// Step 0: retrieve info
+async fn step0(addr: BackendInvitationAddr) -> anyhow::Result<UserOrDeviceClaimInitialCtx> {
+    let handle = start_spinner("Retrieving invitation info");
+
+    let ctx = claimer_retrieve_info(Arc::new(ClientConfig::default().into()), addr).await?;
+
+    handle.done();
+
+    Ok(ctx)
+}
+
+/// Step 1: wait peer
+async fn step1_user(ctx: UserClaimInitialCtx) -> anyhow::Result<UserClaimInProgress1Ctx> {
+    println!(
+        "Invitation greeter: {YELLOW}{}{RESET}",
+        ctx.greeter_human_handle()
+    );
+
+    let handle = start_spinner("Initializing connection with greeter for claiming user");
```

Comment:
I would recommend the following suggestion:

```suggestion
    let handle = start_spinner("Waiting the greeter to start the invitation procedure for an user");
```

Here what I would like the message to convey is:

> We are currently waiting for the greeter to start the procedure

With the current message, the user may think that's the connection that take time to set-up

Reply:
`for an user` ? the user is who reads the message :thinking:

Reply:
We could omit that part then ;)